### PR TITLE
feat(IoTDA): add a resource to manage device certificate

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add a resource t o manage device certificate

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/iotda' TESTARGS='-run=TestAccDeviceCertificate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run=TestAccDeviceCertificate_basic -timeout 360m -parallel 4
=== RUN   TestAccDeviceCertificate_basic
--- PASS: TestAccDeviceCertificate_basic (14.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     14.745s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_certificate_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+func TestAccDeviceCertificate_basic(t *testing.T) {
+	var obj model.CertificatesRspDto
+	rName := "huaweicloud_iotda_device_certificate.test"
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getDeviceCertificateResourceFunc,
+	)
+
+	resource.Test(t, resource.TestCase{
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_certificate_test.go]: please use ParallelTest

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"cn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"owner": {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go:58]: this attribute is missing in the doc

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	var marker *string
+	for {
+		resp, err := client.ListCertificates(&model.ListCertificatesRequest{
+			AppId:  spaceId,
+			Limit:  utils.Int32(50),
+			Marker: marker,
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving IoTDA device certificate, id=%s: %s", id, err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go]: duplicate error message with CheckDeletedDiag

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_certificate_test.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_certificate_test.go]: ok

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go:58]: added

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go]: removed

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+		for _, v := range *resp.Certificates {
+			if utils.StringValue(v.CertificateId) == id {
+				return &v, nil
+			}
+		}
+		marker = resp.Page.Marker
+	}
+
+	return nil, golangsdk.ErrDefault404{}
```
Comment (ShiChangkuo) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go]: suggest to use github.com/huaweicloud/huaweicloud-sdk-go-v3/core/sdkerr.ServiceResponseError

File: docs/resources/iotda_device_certificate.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+* `verify_content` - (Optional, String) Specifies the content of verification certificate. Used to verify the validity
+of the device CA certificate. Get the verification certificate,
+please following [reference](https://support.huaweicloud.com/usermanual-iothub/iot_01_0106.html)
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID in UUID format.
```
Comment (niuzhenguo) [inline docs/resources/iotda_device_certificate.md]: seems it's not UUID format as the import example at L63

File: docs/resources/iotda_device_certificate.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+* `content` - (Required, String, ForceNew) Specifies the content of device CA certificate.
+Changing this parameter will create a new resource. Create a private CA certificate,
+please following [reference](https://support.huaweicloud.com/usermanual-iothub/iot_01_0104.html)
+
+* `space_id` - (Optional, String, ForceNew) Specifies the resource space ID to which the device CA certificate belongs.
+If omitted, the certificate will belong to the default resource space.
+Changing this parameter will create a new resource.
+
+* `verify_content` - (Optional, String) Specifies the content of verification certificate. Used to verify the validity
```
Comment (niuzhenguo) [inline docs/resources/iotda_device_certificate.md]: suggest to add some more description of this verify_content about the limitation. or maybe we just remove this parameter, it seems not suitable for terraform.

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_device_certificate.go]: ok

File: docs/resources/iotda_device_certificate.md (hunk 5 above)
Comment (hmbig2) [reply docs/resources/iotda_device_certificate.md]: fix

File: docs/resources/iotda_device_certificate.md (hunk 6 above)
Comment (hmbig2) [reply docs/resources/iotda_device_certificate.md]: add more description

**Pull request merged**